SearchResults.js

```diff
@@ -0,0 +1,181 @@
+class SearchResult {
+  constructor(searchListId) {
+    this.baseUrl = `https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com/api/v3`;
+    this.searchInput =  document.getElementById("globalSearchInput");
+    this.searchButton = document.getElementById("globalSearchBtn");
+    this.searchList = searchListId;
+    this.setupListeners();
+  }
+
+  debounce(func, delay) {
+    let timeoutId;
+    return function() {
+      const context = this;
+      const args = arguments;
+      clearTimeout(timeoutId);
+      timeoutId = setTimeout(() => {
+        searchList.innerHTML = "";
+        func.apply(context, [...args]);
+      }, delay);
+    }; 
+  }
+  handleSearchClick(event) {
+    event.preventDefault();
+    let loader = document.getElementById("loader");
+    loader.classList.remove("disappear");
+    this.getSearchData()
+  }
+  async getSearchData() {
+    const inputValue = this.searchInput.value;
+    const response = await fetch(`${this.baseUrl}/search?query=${inputValue}&amp%3Blimit=10&amp%3Bexchange=NASDAQ`);
+    const data = await response.json();
+    const firstCompanies = data.slice(0, 10);
+
+    firstCompanies.forEach((company) => {
```

Comment:
instead of forEach, use promise.all() here in order to get your responses faster.